Mathlib/RingTheory/Etale/Basic.lean

```diff
@@ -124,22 +135,68 @@ variable [IsLocalization M Rₘ] [IsLocalization (M.map (algebraMap R S)) Sₘ]
 -- Porting note: no longer supported
 -- attribute [local elab_as_elim] Ideal.IsNilpotent.induction_on
 
-theorem FormallyEtale.of_isLocalization : FormallyEtale R Rₘ :=
+theorem of_isLocalization : FormallyEtale R Rₘ :=
   FormallyEtale.iff_unramified_and_smooth.mpr
     ⟨FormallyUnramified.of_isLocalization M, FormallySmooth.of_isLocalization M⟩
 #align algebra.formally_etale.of_is_localization Algebra.FormallyEtale.of_isLocalization
 
-theorem FormallyEtale.localization_base [FormallyEtale R Sₘ] : FormallyEtale Rₘ Sₘ :=
+theorem localization_base [FormallyEtale R Sₘ] : FormallyEtale Rₘ Sₘ :=
   FormallyEtale.iff_unramified_and_smooth.mpr
     ⟨FormallyUnramified.localization_base M, FormallySmooth.localization_base M⟩
 #align algebra.formally_etale.localization_base Algebra.FormallyEtale.localization_base
 
-theorem FormallyEtale.localization_map [FormallyEtale R S] : FormallyEtale Rₘ Sₘ := by
+theorem localization_map [FormallyEtale R S] : FormallyEtale Rₘ Sₘ := by
   haveI : FormallyEtale S Sₘ := FormallyEtale.of_isLocalization (M.map (algebraMap R S))
   haveI : FormallyEtale R Sₘ := FormallyEtale.comp R S Sₘ
   exact FormallyEtale.localization_base M
 #align algebra.formally_etale.localization_map Algebra.FormallyEtale.localization_map
 
 end Localization
 
+end FormallyEtale
+
+section
+
+variable (R : Type u) [CommSemiring R]
+variable (A : Type u) [Semiring A] [Algebra R A]
+
+/-- An `R`-algebra `A` is étale if it is formally étale and of finite presentation. -/
+class Etale : Prop where
+  formallyEtale : FormallyEtale R A
+  finitePresentation : FinitePresentation R A
+
+end
+
+namespace Etale
+
+attribute [instance] formallyEtale
```

Comment:
Maybe merge master and make `finitePresentation` and instance as well? Same for the other two.